packages/next/src/lib/typescript/runTypeCheck.ts

```diff
@@ -85,7 +85,7 @@ export async function runTypeCheck(
   // - pages/test/a.js
   //
   const regexIgnoredFile =
-    /[\\/]__(?:tests|mocks)__[\\/]|(?<=[\\/.])(?:spec|test)\.[^\\/]+$/
+    /[\\/]__(?:tests|mocks)__[\\/]|(?<=[\\/.])(?:spec|test)\.[^\\/]+$|^.*\/app\/_.*$/
```

Comment:
I think it should skip not only `app/_` but `app/**/_` as they’re all private directories.

Reply:
Great catch, thank you!!